validation/src/test/scala/hmda/validation/rules/ts/validity/V155Spec.scala

```diff
@@ -12,15 +12,17 @@ class V155Spec extends PropSpec with PropertyChecks with MustMatchers with TsGen
   property("A valid email address for the institution must be provided") {
     forAll(tsGen) { ts =>
       whenever(ts.contact.email != "") {
-        V155(ts.contact) mustBe Success()
+        V155(ts) mustBe Success()
       }
     }
   }
 
   property("Invalid emails should be detected") {
-    forAll(badContactGen) { contact =>
-      whenever(contact.email != "") {
-        V155(contact) mustBe Failure("is not a valid email")
+    forAll(tsGen) { ts =>
+      val badContact = badContactGen.sample.getOrElse(Contact("", "", "", ""))
+      val badTs = ts.copy(contact = badContact)
+      whenever(badTs.contact.email != "") {
+        V155(badTs) mustBe Failure("is not a valid email")
```

Comment:
Regarding `2`, this is only half true. The generator always generates a non empty Contact with some email string in it, that is correct. The `badContactGen.sample` at line 22 call returns a `Some(contact)` or a `None`. When `None`, the `getOrElse` call chooses an empty Contact. This might take a few runs to have Scalacheck cover that use case, but it does happen.